srplasticity/srp_extrafuncs.py

```diff
@@ -390,91 +498,140 @@ def plot_srp(model, target_dict, stimulus_dict, protocols=None):
                     marker="o",
                     markersize=2,
                     label="Data",
+                    capsize=2,
                 )
             fig.panels[ix].plot(xax, mean, color="#cc3311", label="SRP model")
+            fig.panels[ix].fill_between(xax, lower_SD, upper_SD, color="xkcd:light grey", label="SD")
             if protocols != None:
                 fig.panels[ix].set_title(protocols[key])
             else:
                 fig.panels[ix].set_title(key)
-            fig.panels[ix].set_xticks(xax)
+            if len(xax) <= 10:
+                fig.panels[ix].set_xticks(xax)
+            else:
+                ticks = np.arange(1, len(mean) + 1, math.ceil(len(mean) / 10))
+                fig.panels[ix].set_xticks(ticks)
             fig.panels[ix].set_ylim(0.5, 9)
-            fig.panels[ix].set_yticks([1, 3, 5, 7, 9])
+            fig.panels[ix].set_yticks([1, 3, 5, 7, 9, 11])
 
         fig.panels[0].legend(frameon=False)
         fig.panels[0].set_ylabel("norm. EPSC amplitude")
 
         plt.show()
 
 
-def plot_estimates(means):
+def plot_estimates(means, efficacies):
+    """
+    Plot sample estimates over time with the corresponding mean values.
+
+    :param means: A list of mean predicted responses
+    :type means: list
+    :param efficacies: A list of sample predicted responses
+    :type efficacies: list
+    """
+
     fig, axis = plt.subplots()
-    xax = range(len(means))
-    axis.plot(range(1, len(xax) + 1), means, lw=1, color="#cc3311")
+
+    if type(efficacies[0]) != np.float64 and type(efficacies[0]) != float:
```

Comment:
This is not an ideal Pythonic way to perform type comparisons. Type comparisons are also somewhat problematic to begin with: http://www.canonical.org/~kragen/isinstance/